File: 0283-move-zeroes/0283-move-zeroes.py
class Solution:
    def moveZeroes(self, nums: List[int]) -> None:
        """
        Do not return anything, modify nums in-place instead.
        """
        
        l = 0
        for r in range(len(nums)):
            if nums[l] == 0 and nums[r] != 0:
                nums[l], nums[r] = nums[r] , nums[l]
            
            while nums[l] != 0 and l < r:
                l += 1

        
            


        

File: 0283-move-zeroes/0283-move-zeroes.js
/**
 * @param {number[]} nums
 * @return {void} Do not return anything, modify nums in-place instead.
 */
var moveZeroes = function(nums) {

    let p1 = 0;
    for(let p2=0; p2<nums.length; p2++){
        if(nums[p2] != 0 ){
            [nums[p2] , nums[p1]] = [nums[p1], nums[p2]]
            p1++;
        }
    }  
};